packages/http-client/src/ceramic-http-client.ts

```diff
@@ -135,6 +137,8 @@ export default class CeramicClient implements CeramicApi {
     const docIdStr = docId.toString()
     if (!this._docmap[docIdStr]) {
       this._docmap[docIdStr] = await Document.load(docId, this._apiUrl, this.context, this._config)
+    } else {
+      this._docmap[docIdStr]._syncState()
```

Comment:
At this point is there even any reason to be caching the documents at all?

Also I think you need `await`

Reply:
same as above. Nice catch with `await`.

Reply:
Ah interesting, so this isn't really a cache at all, it's just a way to hold on to the handle to Documents that have been handed out so we can update the state out from under them.  Interesting.  Would be nice to write that down somewhere, can we add a comment to _docmap?

Reply:
What would you include in that comment?

Reply:
How about:
`_docmap stores handles to Documents that been handed out.  This allows us to update the state within the Document object when we learn about changes to the document.  This means that client code with Document references always have access to the most recent known-about version, without needing to explicitly re-load the document.`

Reply:
Thanks, added.